### DISCUSSION TITLE
Azure Active Directory `redirect_uri=http`

### DISCUSSION THREAD

**Discussion by Mike-E-angelo** [datalust/seq-tickets#1438] [Help] [answered]
I am setting up Seq using this post here:
https://ml-software.ch/posts/running-seq-on-azure

That post worked out well for me and I can now access my globally-available Seq instance, which is awesome!  Excitement aside, I have set out with my first task of setting up Azure Active Directory authentication.

In doing so, I have supplied all the necessary values, and the return URL in the setup page reads `https://<azuresite-url>/aad` as seen here:

![](https://i.imgur.com/IR8ku1Z.png)

However, when I actually attempt to login, the `redirect_uri` as a query parameter is `http` (and not `https`).

This is leading to the same problem described here:
https://docs.datalust.co/v4/discuss/5e84a05bf58b750065b047b5

This recommends setting an environment variable.  However, it is unclear how to do so. In the post above there are "application settings" which I added `BASE_ADDRESS` followed by the desired base address.  Even so (and after restarting a few times), the problem persists.

Additionally, I have tried the following, and it does not seem to work:
https://stackoverflow.com/a/50715664/10340424

So unfortunately, it seems the return URL is correctly listed as `https` when I enable AAD in the settings page, but not when I actually sign in.

Any additional pointers here would be greatly appreciated.

**Comment by nblumhardt** [comment, marked as answer]
Hi Mike,

Thanks for your message. Unfortunately, the variable name is actually `BASE_URI`, though you might try `SEQ_API_CANONICALURI`, which is the newer name for this setting.

I'll post some comments to the same effect on the blog post - since that one was written there are a few more variables that we recommend setting on the initial deployment to Azure:

 * `SEQ_API_CANONICALURI` - the public https:// URL for the Seq server
 * `SEQ_FIRSTRUN_ADMINUSERNAME` - a username for the default admin user (`admin`, if not specified)
 * `SEQ_FIRSTRUN_ADMINPASSWORDHASH` - a hash of the admin user's initial password, generated by typing into `seq config hash`
 * ~~`SEQ_FEATURES_ENABLED` - the value should be `nativemetastore` to use the new metadata store that we've made some improvements to for Azure (Premium) Files storage~~ (edit: no longer required for recent Seq versions)

Last thing - Azure Files seems to be prone to corruptions with Seq's workload; we recommend Azure Premium Files for even moderately-loaded servers. As one commenter points out, VMs with managed disks are still the "serious" Azure option, but Premium Files gets closer to what Seq needs for reliable operation.

HTH! Keen to hear how you to :-)

**Reply by Mike-E-angelo** [reply]
WOOHOO!  Thank you very much for the reply @nblumhardt!  I am not sure which one it was, but I put both of those variables into my Application Settings, and I now have access!  SOOOOOOOOO NICE!

I will look into Azure Premium Files and get that setup.  This is for "hello world"/preview of my application, so baby steps for now. :)  Thank you again for your assistance and great work over there!
